src/transformers/trainer_pt_utils.py

```diff
@@ -55,6 +55,15 @@
 logger = logging.get_logger(__name__)
 
 
+def get_dataloader_sampler(dataloader):
+    # after accelerate.prepare function the wraped dataloader.sampler will be SequentialSampler instead of RandomSampler
+    if hasattr(dataloader, "sampler") and isinstance(dataloader.sampler, RandomSampler):
+        return dataloader.sampler
+    if hasattr(dataloader, "batch_sampler") and dataloader.batch_sampler is not None:
+        return get_dataloader_sampler(dataloader.batch_sampler)
+    return dataloader.sampler
```

Comment:
* In the last line, you can't return `dataloader.sampler` if there dataloader doesn't have that attribute. 
* The `isinstance` check isn't needed. You just want to return any sampler and then the user can test if they want. In fact, the current logic will already do that: it would return a random sample because it would evaluate as false for the if checks and then finally return `dataloader.sampler`

```suggestion
    if hasattr(dataloader, "sampler"):
        return dataloader.sampler
    if hasattr(dataloader, "batch_sampler") and dataloader.batch_sampler is not None:
        return get_dataloader_sampler(dataloader.batch_sampler)
```